web/packages/teleterm/src/ui/DocumentTerminal/useDocumentTerminal.ts

```diff
@@ -53,12 +54,25 @@ export function useDocumentTerminal(doc: types.DocumentTerminal) {
       documentsService.update(doc.uri, { status: 'connecting' });
     }
 
+    // Add `shellId` before going further.
+    let docWithDefaultShell: types.DocumentTerminal;
+    if (
+      (doc.kind === 'doc.terminal_shell' || doc.kind === 'doc.gateway_kube') &&
+      !doc.shellId
+    ) {
+      docWithDefaultShell = {
+        ...doc,
+        shellId: ctx.configService.get('terminal.shell').value,
+      };
+      documentsService.update(doc.uri, docWithDefaultShell);
+    }
```

Comment:
>Is this to update any existing documents which do not have this field set? 

No, the purpose of it was to add a default shell  to any new "shell" document (in `DocumentsService` we set `shellId: ''`).
That `shellId` is needed later in `useDocumentTerminal` and in the context menu, to mark the correct option.
The problem was that `DocumentsService` depends only on some setters and getters from `WorkspacesService`. I wasn't sure about adding `ConfigService` there (which we would call in `newTerminalDocument`), so I came up with this late initialization.
But actually, after we added returning the resolved shell from `buildPtyOptions` (and updating the document with it), we can get rid of it. This should be probably even more correct (than updating the document with the default shell), because we don't actually know what shell we will get from `buildPtyOptions`, so we can update the document after we get it.